parsing/parsetree.mli

```diff
@@ -165,541 +182,584 @@ and row_field = {
+   - [{ ...; l: T; ... }]
+                           when {{!label_declaration.pld_mutable}[pld_mutable]}
+                             is {{!Asttypes.mutable_flag.Immutable}[Immutable]},
+   - [{ ...; mutable l: T; ... }]
+                           when {{!label_declaration.pld_mutable}[pld_mutable]}
+                             is {{!Asttypes.mutable_flag.Mutable}[Mutable]}.
+
+   Note: [T] can be a {{!core_type_desc.Ptyp_poly}[Ptyp_poly]}.
 *)
 
-and constructor_declaration =
-    {
-     pcd_name: string loc;
-     pcd_vars: string loc list;
-     pcd_args: constructor_arguments;
-     pcd_res: core_type option;
-     pcd_loc: Location.t;
-     pcd_attributes: attributes; (* C of ... [@id1] [@id2] *)
-    }
+and constructor_declaration = {
+  pcd_name : string loc;
+  pcd_vars : string loc list;
+  pcd_args : constructor_arguments;
+  pcd_res : core_type option;
+  pcd_loc : Location.t;
+  pcd_attributes : attributes;  (** [C of ... [\@id1] [\@id2]] *)
+}
 
 and constructor_arguments =
   | Pcstr_tuple of core_type list
   | Pcstr_record of label_declaration list
-
-(*
-  | C of T1 * ... * Tn     (res = None,    args = Pcstr_tuple [])
-  | C: T0                  (res = Some T0, args = [])
-  | C: T1 * ... * Tn -> T0 (res = Some T0, args = Pcstr_tuple)
-  | C of {...}             (res = None,    args = Pcstr_record)
-  | C: {...} -> T0         (res = Some T0, args = Pcstr_record)
-  | C of {...} as t        (res = None,    args = Pcstr_record)
+      (** Values of type {!constructor_declaration}
+    represents the constructor arguments of:
+  - [C of T1 * ... * Tn]     when [res = None],
+                              and [args = Pcstr_tuple [T1; ... ; Tn]],
+  - [C: T0]                  when [res = Some T0],
+                              and [args = Pcstr_tuple []],
+  - [C: T1 * ... * Tn -> T0] when [res = Some T0],
+                              and [args = Pcstr_tuple [T1; ... ; Tn]],
+  - [C of {...}]             when [res = None],
+                              and [args = Pcstr_record [...]],
+  - [C: {...} -> T0]         when [res = Some T0],
+                              and [args = Pcstr_record [...]],
+  - [C of {...} as t]        when [res = None],
+                              and [args = Pcstr_record [...]].
 *)
 
-and type_extension =
-    {
-     ptyext_path: Longident.t loc;
-     ptyext_params: (core_type * (variance * injectivity)) list;
-     ptyext_constructors: extension_constructor list;
-     ptyext_private: private_flag;
-     ptyext_loc: Location.t;
-     ptyext_attributes: attributes;   (* ... [@@id1] [@@id2] *)
-    }
-(*
-  type t += ...
+and type_extension = {
+  ptyext_path : Longident.t loc;
+  ptyext_params : (core_type * (variance * injectivity)) list;
+  ptyext_constructors : extension_constructor list;
+  ptyext_private : private_flag;
+  ptyext_loc : Location.t;
+  ptyext_attributes : attributes;  (** ... [\@\@id1] [\@\@id2] *)
+}
+(**
+  Values of type [type_extension] represents [type t += ...].
```

Comment:
I would use a less neutral form:
```
Definition of new extensions constructors for the extensive sum type [t]
```
(And it might be useful to comment each field, but let's do that later).